webdriver-spec.html

```diff
@@ -4581,14 +4583,35 @@
  the following steps need to be completed:
 
 <ol>
- <li><p>Let <var>result</var> be Return the result of
+ <li><p>Let <var>evaluateResult</var> be the result of
   calling <a><code>evaluate</code></a>, with
-  arguments <var>selector</var>, <a>context object</a> equal to
-  the <var>start node</var>.
+  arguments <var>selector</var>, <var>start node</var>,
+  <a><code>null</code></a>, <a>ORDERED_NODE_SNAPSHOT_TYPE</a>, and
+  <a><code>null</code></a>.
+
+  <p class=note>A snapshot is used to promote operation atomicity.
+
+ <li><p>Let <var>index</var> be 0.
+
+ <li><p>Let <var>length</var> be the result of <a>getting the property</a>
+  "<code>snapshotLength</code>" from <var>evaluateResult</var>.
+
+ <li><p>Let <var>result</var> be an empty <a>NodeList</a>.
 
- <li><p>If any item in <var>result</var> is not an <a>element</a>
-  return an <a>error</a> with <a>error code</a> <a>invalid
-  selector</a>.
+ <li><p>Repeat, while <var>index</var> is less than <var>length</var>:
+
+  <ol>
+   <li>Let <var>node</var> be the result of calling <a>snapshotItem</a> with
+    argument <var>index</var>, with the <a>context object</a> equal to
+    <var>evaluateResult</var>.
+
+   <li><p>If <var>node</var> is not an <a>element</a> return an <a>error</a>
+    with <a>error code</a> <a>invalid selector</a>.
```

Comment:
I think we'll be fine with updating the `Find` algorithm's step 5 to read:

```
Let elements returned be the result of *trying* the relevant element location strategy call with arguments start node, and selector.
```

That is, to make use of our `trying` mechanism for fallible algorithms, where the handling is already clear.